File: Assignment3/C/Program4.c
// PS - Write a program which accept number from user and display its summation of all Non - factors

#include<stdio.h>

int NonFactSum(int iNo){
    int iCnt = 0;
    int iSum = 0;

    for(iCnt = 1; iCnt <= (iNo/2); iCnt++){
        if((iNo % iCnt) != 0) {
            iSum = iSum + iCnt;
        }
    }
    return iSum;
}

int main() {
    int iValue = 0;
    int iRet = 0;

    printf("Enter number \n");
    scanf("%d", &iValue);

    iRet = NonFactSum(iValue);

    printf("SUmmation of non-factors is %d \n", iRet);

    return 0;
}

File: Assignment34/Java/Program4.java
// PS - Write a java program which accept N numbers from user and display all such numbers which are even and divisible by 5

import java.util.*;

class ArrayX
{
    public void DivisibleBy5andEven(int Arr[])
    {
        for(int iCnt = 0; iCnt< Arr.length; iCnt++)
        {
            if(((Arr[iCnt] % 2) == 0) && ((Arr[iCnt] % 5) == 0))
            {
                System.out.print(Arr[iCnt]+"\t");
            }
        }
    }
}

class Program4
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        int iSize = 0;
        System.out.println("Enter number of elements ");
        iSize = sobj.nextInt();

        int Arr[] = new int[iSize];

        System.out.println("Enter elements ");
        for(int iCnt = 0; iCnt < iSize; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }

        ArrayX aobj = new ArrayX();
        aobj.DivisibleBy5andEven(Arr);

    }
}